lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -483,6 +485,15 @@ newDBLayer trace fp timeInterpreter = do
 
         removePools = mapM_ $ \pool -> do
             liftIO $ traceWith trace $ MsgRemovingPool pool
+            -- Remove all metadata associated with the pool, bearing in mind
+            -- that a single pool can have several historical metadata hashes
+            -- associated with it:
+            metadataHashesToRemove <-
+                mapMaybe (poolRegistrationMetadataHash . entityVal) <$>
+                selectList [ PoolRegistrationPoolId ==. pool ] []
+            forM_ metadataHashesToRemove $ \metadataHash ->
+                deleteWhere [ PoolMetadataHash ==. metadataHash ]
```

Comment:
Hmmm. That is actually not okay ^^"
Multiple pools could have the same metadata and same metadata hash. This would open up attacks such that pool B registers after pool A with the same metadata, and immediately de-register; clearing up pool's A metadata and making it disappear from the listing.

I suggest instead to add a proper comment to that `removePools` function explaining why metadata are not cleaned up and treated as an immutable data-store.

Reply:
@KtorZ wrote:
> Multiple pools could have the same metadata and same metadata hash.
> This would open up attacks such that pool B registers after pool A with the same metadata, and immediately de-register; clearing up pool's A metadata and making it disappear from the listing.

That's a really good point.

> I suggest instead to add a proper comment to that `removePools` function explaining why metadata are not cleaned up and treated as an immutable data-store.

How about only removing metadata when it is _no longer reachable_?

When removing pool **_p_** with metadata hash **_h_**:

1. First remove all rows corresponding to pool **_p_** from `pool_registrations` and `pool_retirements`, as before.
2. Now find the count **_c_** of all rows in `pool_registrations` with `metadata_hash` = **_h_**.
3. If **_c_** **= 0**, then remove all entries from `pool_metadata` with `metadata_hash` == **_h_**, otherwise leave this table _untouched_.

Reply:
After some discussion, it was decided that we will not fix this (for the moment).

Strictly speaking, when removing pools, we should also be able to remove any pool metadata that becomes unreachable as a result (as described in the [above comment](https://github.com/input-output-hk/cardano-wallet/pull/2160#discussion_r492565691)).

However, for the moment, to keep things simple, we adopt the principle of never removing metadata. We can revise this decision in future if the metadata table grows too large.